feat(HTTP): Refresh 2XX response pages (#34422)
mdn/content: files/en-us/web/http/status/200/index.md

200 OK
```

## Examples

### GET response

Making a successful GET request to `https://example.com` returns the following response with the HTML contained in the message body:

```http
HTTP/1.1 200 OK
Accept-Ranges: bytes
Age: 294510
Cache-Control: max-age=604800
Content-Type: text/html; charset=UTF-8
Date: Fri, 21 Jun 2024 14:18:33 GMT
Etag: "3147526947"
Expires: Fri, 28 Jun 2024 14:18:33 GMT
Last-Modified: Thu, 17 Oct 2019 07:18:26 GMT
Server: ECAcc (nyd/D10E)
X-Cache: HIT
Content-Length: 1256

<!doctype html>
```

### POST response in form submission

Assuming a form exists to send data to an endpoint for managing subscriptions at `http://example.com/subscribe`.
A `POST` request to subscribe a user may look like the following:

```http
POST /subscribe HTTP/1.1
Host: example.com
Content-Type: application/x-www-form-urlencoded
Content-Length: 50

name=Brian%20Smith&email=brian.smith%40example.com
```

An example response could look like this:

```http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "message": "User subscription pending. A confirmation email has been sent.",
  "subscription": {
    "name": "Brian Smith",
    "email": "brian.smith@example.com",
    "id": 123,

>     "feedNumber": "default"

Author: ```suggestion
    "feed": "default"
```

{{HTTPSidebar}}

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Author: Good idea, what about this sentence order:

```
what it is. (200 means the request was successful)
what class it's in. (2XX range)
notes (if it's cacheable / other)
```

I want to bring the class (category) pretty high up in page information.

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Reviewer: This pattern looks good, thanks!

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Reviewer: 1. The issue is that "is part of" is breaking the flow of info about the response code:

- "The HTTP 200 OK status response code indicates that a request has succeeded. It is part of the 200-299 class of successful responses. A 200 OK response is cacheable by default."

vs

- "The HTTP 200 OK status response code indicates that a request has succeeded. A 200 OK response is cacheable by default. It is part of the 200-299 class of successful responses."

I personally prefer the second version because it presents information in the order of first describing what it is, followed by any other essential info, and then additional useful info. Though important, I regard the category info secondary and to be mentioned only after the core details.

2. Similarly, in cases like [202](https://pr34422.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTTP/Status/202), the current order of sentences is breaking the flow of information.

- "The HTTP 202 Accepted status response code indicates that a request has been accepted for processing, but processing has not been completed or may not have started. It is part of the 200-299 class of [successful responses](https://pr34422.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTTP/Status#successful_responses).

    The actual processing of the request is not guaranteed; a task or action may fail or be disallowed when a server tries to process it. A 202 response is non-committal, meaning there is no way to later send an asynchronous HTTP response to indicate the outcome of the processing. This response code is typically used when the request is handled by another process or server, or when requests are processed in batches."

- I'd reorganize it as:

"The HTTP 202 Accepted status response code indicates that a request has been accepted for processing, but processing has not been completed or may not have started. The actual processing of the request is not guaranteed; a task or action may fail or be disallowed when a server tries to process it. A 202 response is non-committal, meaning there is no way to later send an asynchronous HTTP response to indicate the outcome of the processing. This response code is typically used when the request is handled by another process or server, or when requests are processed in batches. It is part of the 200-299 class of successful responses."

OR if we could add a "Description" section, we could move the addiitonal info there and keep the intro short, so:

```
The HTTP 202 Accepted status response code indicates that a request has been accepted
for processing, but processing has not been completed or may not have started. This
response code is typically used when the request is handled by another process or server,
or when requests are processed in batches. It is part of the 200-299 class of successful responses.

## Description

The actual processing of the request is not guaranteed; a task or action may fail or be
disallowed when a server tries to process it. A 202 response is non-committal, meaning there
is no way to later send an asynchronous HTTP response to indicate the outcome of the processing. 
```

LMK what you think. If you're not too keen with this suggestion, it might be better to revert to your original sentence structure, that is,  "The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates..." because it doesnt break the flow; I suggested the change so we could present more important info upfront.

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Author: Thanks, Dipika!

>The issue is that "is part of" is breaking the flow of info about the response code

I agree, we're doing some hard work on some pages including the sentence as a standalone in different spots and it is a stopper in some cases. What we do on API pages is also a good convention:

```md
The **`SyncEvent`** interface of the {{domxref("Background Synchronization API")}} represents a sync action that is dispatched on the {{domxref("ServiceWorkerGlobalScope")}} of a ServiceWorker.
```

So we have:

>A **`this feature`** of the [parent or category](/path/to/parent) thing represents a ...

So we could move the original suggestion into the first sentence like:

* The HTTP **`200 OK`** [successful response](/en-US/docs/Web/HTTP/Status#successful_responses) status code indicates...

or

* The **`200 OK`** HTTP response is a [successful response status code](/en-US/docs/Web/HTTP/Status#successful_responses) that indicates...

Then we have a link to response class right away. We don't need to mention the number range, I guess it's too much, and that will be apparent when the reader clicks the "successful response" bit.

I think in general a Description section is not necessary as the pages are so short, the first two paras are the description, really, so we can optimize those instead! What do you reckon?

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Author: What do you think about this: https://github.com/mdn/content/compare/main...bsmth:content:http-3xx-responses

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Author: Do you want to check https://github.com/mdn/content/pull/34422/commits/f5c2080fb13ae8b7335cc3c05d864776645f934f? I think it's a good compromise for linking to the category while staying out of the way of the vital information, what do you think?

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Reviewer: "The HTTP 200 OK [successful response](https://github.com/en-US/docs/Web/HTTP/Status#successful_responses) status code indicates..." is indeed a very neat solution, providing all the info whilst not breaking the flow! Let's go with this format. Thank you!!

> I think in general a Description section is not necessary as the pages are so short, the first two paras are the description, really, so we can optimize those instead! What do you reckon?
I agree, no separate section required if it's just two short paras but also depends on the type of content; I would consider moving additional content to a separate "Description" section if the content starts explaining the mechanics of processing and if there are additional caveats (notes).

> The HTTP **`200 OK`** status response code is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses) and indicates that a request has succeeded.

Author: Super, thank you!

A 200 response is cacheable by default.

> The meaning of success depends on the HTTP request method used:

Author: Yeah the latter "The interpretation of a request being 'successful' depends on.."

> - {{HTTPMethod("GET")}}: The resource has been retrieved and is included in the response body.

Author: I've gone with some other wording to keep it brief, do you want to have another look?

> - {{HTTPMethod("GET")}}: The resource has been retrieved and is included in the response body.

Reviewer: The rewording looks much better. Thanks! Adding a few nits

> - {{HTTPMethod("GET")}}: The resource has been retrieved and is included in the response body.

Author: Cool ty

- {{HTTPMethod("HEAD")}}: The response has representation headers, but no message body.
- {{HTTPMethod("POST")}}: The response has a message body that describes the result of the action.

> - {{HTTPMethod("TRACE")}}: The response has a message body with the request message as received by the server.

Author: Might want to check again after some rewording

> The successful result of a {{HTTPMethod("PUT")}} or a {{HTTPMethod("DELETE")}} is often not `200 OK`, but {{HTTPStatus("204")}} `No Content`, or {{HTTPStatus("201")}} `Created` if the resource is uploaded for the first time.

Author: I think it's okay to have it separate as we're talking about methods you might expect a 200 for, but may respond with something else.

> The successful result of a {{HTTPMethod("PUT")}} or a {{HTTPMethod("DELETE")}} is often not `200 OK`, but {{HTTPStatus("204")}} `No Content`, or {{HTTPStatus("201")}} `Created` if the resource is uploaded for the first time.

Reviewer: The new write up looks good

## Status

```http
200 OK
```

## Examples

### GET response

> Making a successful GET request to `https://example.com` returns the following response with the HTML contained in the message body:

Reviewer: also then, we can present this as the example scenario
```suggestion
In this example, a successful GET request to `https://example.com` returns a `200 OK` response. The response includes ...
```
should we say includes HTTP headers (as shown here), which is followed by the message body?

```http
HTTP/1.1 200 OK
Accept-Ranges: bytes
Age: 294510
Cache-Control: max-age=604800
Content-Type: text/html; charset=UTF-8
Date: Fri, 21 Jun 2024 14:18:33 GMT
Etag: "3147526947"
Expires: Fri, 28 Jun 2024 14:18:33 GMT
Last-Modified: Thu, 17 Oct 2019 07:18:26 GMT
Server: ECAcc (nyd/D10E)
X-Cache: HIT
Content-Length: 1256

> <!doctype html>

Reviewer: should we add this comment:
```suggestion
<!doctype html>
<!-- HTML content will follow here -->
```

Author: Yeah let's add

```

### POST response in form submission

Assuming a form exists to send data to an endpoint for managing subscriptions at `http://example.com/subscribe`.
A `POST` request to subscribe a user may look like the following:

```http
POST /subscribe HTTP/1.1
Host: example.com
Content-Type: application/x-www-form-urlencoded
Content-Length: 50

name=Brian%20Smith&email=brian.smith%40example.com
```

> An example response could look like this:

Reviewer: ```suggestion
In this example, a successful response with a `200 OK` status could look like this:
```

```http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "message": "User subscription pending. A confirmation email has been sent.",
  "subscription": {
    "name": "Brian Smith",
    "email": "brian.smith@example.com",
    "id": 123,
    "feed": "default"
  }
}
```

## Specifications

{{Specifications}}

## See also

- [HTTP request methods](/en-US/docs/Web/HTTP/Methods)
- [HTTP response status codes](/en-US/docs/Web/HTTP/Status)

> - {{Glossary("Idempotent")}} Glossary entry

Reviewer: how about
```suggestion
- Glossary: {{Glossary("Idempotent")}}
```

{{HTTPSidebar}}

The HTTP **`200 OK`** status response code indicates that a request has succeeded.
It is part of the `200`-`299` class of [successful responses](/en-US/docs/Web/HTTP/Status#successful_responses).
A `200 OK` response is cacheable by default.

> A `200 OK` has the following meaning and format in response to these request methods:

Reviewer: Sorry, coming back to this one. My problem is with the placement of "following". Ideally, we'd want to say "following methods" because that's what the list contains.
How about rephrasing to something like:
```suggestion
The meaning of a `200 OK`, indicating a successful request, depends on the following HTTP request methods:
```

Author: Yeah I see what you mean, I'm coming from the perspective that a 200 looks different depending on the request method. So 'success' is different each time and the response has a different format depending on the request method.

Author: Your suggestion is good, we should add that the format (whether there's a body) is different, also.

Author: Alternatives:

* "A `200 OK` has a different meaning and format depending on the request method:"
* "The meaning of 'successful', and the format of the 200 responses is different depending on the request method that the server is responding to:"

Reviewer: Gotcha, going with your previous version :) with a tweak:

```suggestion
A `200 OK` response has a different meaning and format depending on the HTTP request method. Here's how they vary for different methods:
```

Author: great ty

> - {{HTTPMethod("GET")}}: A resource was retrieved by the server and it's included in the response body.
> - {{HTTPMethod("POST")}}: An action succeeded, the response has a message body describing the result.
> - {{HTTPMethod("HEAD")}}: Identical to `GET`, except there is no message body.
> - {{HTTPMethod("TRACE")}}: The response has a message body containing the request as received by the server.

Reviewer: ```suggestion
- {{HTTPMethod("GET")}}: A resource was retrieved by the server and included in the response body.
- {{HTTPMethod("POST")}}: An action succeeded; the response has a message body describing the result.
- {{HTTPMethod("HEAD")}}: Identical to `GET`, except there is no message body.
- {{HTTPMethod("TRACE")}}: The response has a message body containing the request as received by the server.
```

> Although possible, a successful {{HTTPMethod("PUT")}} or {{HTTPMethod("DELETE")}} request often does not result in a `200 OK` response.

Reviewer: ```suggestion
Although possible, successful {{HTTPMethod("PUT")}} or {{HTTPMethod("DELETE")}} requests often do not result in a `200 OK` response.
```

Author: good

It is more common to see {{HTTPStatus("201")}} `Created` if the resource is uploaded or created for the first time, or {{HTTPStatus("204")}} `No Content` upon successful deletion of a resource.

## Status

```http
200 OK
```

## Examples

> ### Receiving a `200 OK` for a GET request
> 
> In this example, a successful GET request to `https://example.com` returns a `200 OK` response.

Reviewer: ```suggestion
### Receiving a `200 OK` for a `GET` request

In this example, a successful `GET` request to `https://example.com` returns a `200 OK` response.
```

> The response includes representation headers and a message body with the HTML resource:

Author: ```suggestion
The response includes representation headers and a message body with the HTML content:
```
